### PR TITLE
Feat/1128/improve cache control

### PR DESCRIPTION
**Pull request by Tomvbe**

### CODE REVIEW

File: docs/_configuration/fragmentations/timebased.md [hunk 1]
```diff
@@ -29,6 +30,37 @@ For maxGranularity the following values are allowed:
...
+timestamp: 21/02/2024
+
+Will result in fragments:
+- root
+- year=2024
+- year=2024&month=02
+- year=2024&month=02&day=21
+
+We will assume that no new year will be added to the root until 31/12/2024.
+We will assume that no new month will be added to year=2024 until 29/02/2024.
```
Comment (Yalz) [inline docs/_configuration/fragmentations/timebased.md:48]: Just to verify, It keeps track of leap years?

File: ldes-fragmentisers/ldes-fragmentisers-common/src/main/java/be/vlaanderen/informatievlaanderen/ldes/server/fragmentation/repository/FragmentRepository.java [hunk 2]
```diff
@@ -58,4 +58,13 @@ Optional<Fragment> retrieveMutableFragment(String viewName,
...
 	void removeRelationsPointingToFragmentAndDeleteFragment(LdesFragmentIdentifier readyForDeletionFragment);
+
+	/**
+	 * Finds and updates all the child relations of the fragment and makes them immutable.
+	 * For example fragment year=2024, all fragments referring to months, days, hours, etc. within 2024 will become immutable.
+	 * The fragment itself does NOT become immutable. Only the children
+	 * When the fragment has no children, or only has immutable children, then no updates are done.
+	 * @param fragment The fragment whose children should become immutable
+	 */
+    void makeChildrenImmutable(Fragment fragment);
```
Comment (Yalz) [inline ldes-fragmentisers/ldes-fragmentisers-common/src/main/java/be/vlaanderen/informatievlaanderen/ldes/server/fragmentation/repository/FragmentRepository.java:69]: tabbing seems off.. ?

File: ldes-fragmentisers/ldes-fragmentisers-timebased-hierarchical/src/main/java/be/vlaanderen/informatievlaanderen/ldes/server/fragmentisers/timebasedhierarchical/HierarchicalTimeBasedFragmentationStrategyWrapper.java [hunk 3]
```diff
@@ -35,7 +35,8 @@ public FragmentationStrategy wrapFragmentationStrategy(ApplicationContext applic
 
 	private TimeBasedConfig createConfig(ConfigProperties properties) {
 		return new TimeBasedConfig(properties.getOrDefault(FRAGMENTATION_SUBJECT_FILTER, ".*"),
-				properties.get(FRAGMENTATION_PATH), Granularity.from(properties.get(MAX_GRANULARITY)));
+				properties.get(FRAGMENTATION_PATH), Granularity.from(properties.get(MAX_GRANULARITY)),
+				Boolean.parseBoolean(properties.getOrDefault(LINEAR_TIME_CACHING_ENABLED, "")));
```
Comment (Yalz) [inline ldes-fragmentisers/ldes-fragmentisers-timebased-hierarchical/src/main/java/be/vlaanderen/informatievlaanderen/ldes/server/fragmentisers/timebasedhierarchical/HierarchicalTimeBasedFragmentationStrategyWrapper.java:39]: I would expect the default to be "false" ?

File: ldes-fragmentisers/ldes-fragmentisers-timebased-hierarchical/src/main/java/be/vlaanderen/informatievlaanderen/ldes/server/fragmentisers/timebasedhierarchical/services/TimeBasedFragmentCreator.java [hunk 4]
```diff
@@ -27,35 +27,47 @@ public TimeBasedFragmentCreator(FragmentRepository fragmentRepository,
...
 
-	public Fragment getOrCreateFragment(Fragment parentFragment, String timeValue,
+	public Fragment getOrCreateFragment(Fragment parentFragment,
+										String timeValue,
 										Granularity granularity) {
-		Fragment child = parentFragment
-				.createChild(new FragmentPair(granularity.getValue(),
-						timeValue));
+
+		// granu
```
Comment (Yalz) [inline ldes-fragmentisers/ldes-fragmentisers-timebased-hierarchical/src/main/java/be/vlaanderen/informatievlaanderen/ldes/server/fragmentisers/timebasedhierarchical/services/TimeBasedFragmentCreator.java:40]: Seems like comment rubble?

**Review by Yalz — APPROVED** [review]

File: docs/_configuration/fragmentations/timebased.md (hunk 1 above)
Comment (Tomvbe) [reply docs/_configuration/fragmentations/timebased.md:48]: Yes, we use the default java.time.temporal.TemporalAdjusters to determine the last day of the month. This library keeps track of leap years.

File: ldes-fragmentisers/ldes-fragmentisers-common/src/main/java/be/vlaanderen/informatievlaanderen/ldes/server/fragmentation/repository/FragmentRepository.java (hunk 2 above)
Comment (Tomvbe) [reply ldes-fragmentisers/ldes-fragmentisers-common/src/main/java/be/vlaanderen/informatievlaanderen/ldes/server/fragmentation/repository/FragmentRepository.java:69]: This must be a github thing. In intellij, this seems fine.

File: ldes-fragmentisers/ldes-fragmentisers-timebased-hierarchical/src/main/java/be/vlaanderen/informatievlaanderen/ldes/server/fragmentisers/timebasedhierarchical/services/TimeBasedFragmentCreator.java (hunk 4 above)
Comment (Tomvbe) [reply ldes-fragmentisers/ldes-fragmentisers-timebased-hierarchical/src/main/java/be/vlaanderen/informatievlaanderen/ldes/server/fragmentisers/timebasedhierarchical/services/TimeBasedFragmentCreator.java:40]: woops, will remove this

**Pull request merged**